dysco compilation error on macos-14 with boost-python3 version 1.85

Hey there,
on macos Sonoma 14.4.1 (arm64), I installed the latest versions of the casacore dependencies, including boost 1.85

```bash
brew install fftw hdf5 boost-python3 numpy cfitsio wcslib wget gfortran gsl ninja
cd ~/src/
git clone https://github.com/casacore/casacore.git
cd casacore
mkdir build
cd build
cmake \
  -G Ninja \
  -DBUILD_TESTING=ON \
  -DCMAKE_PREFIX_PATH=/usr/local/ \
  -DUSE_OPENMP=OFF \
  -DUSE_HDF5=ON \
  -DBUILD_PYTHON=OFF \
  -DBUILD_PYTHON3=ON \
  -DPython3_EXECUTABLE=`which python3` \
  -DBoost_NO_BOOST_CMAKE=True \
  -DCMAKE_Fortran_COMPILER=`which gfortran` \
  -DDATA_DIR=. \
  ..
cmake --build .
```

This is the error I get:

```txt
[1066/1971] Building CXX object tables/Dysco/CMakeFiles/tDysco.dir/tests/testdyscostman.cc.o
FAILED: tables/Dysco/CMakeFiles/tDysco.dir/tests/testdyscostman.cc.o 
/Library/Developer/CommandLineTools/usr/bin/c++ -DCFITSIO_VERSION_MAJOR=4 -DCFITSIO_VERSION_MINOR=400 -DHAVE_DYSCO -DHAVE_FFTW3 -DHAVE_FFTW3_THREADS -DHAVE_HDF5 -DHAVE_READLINE -DUSE_THREADS -DWCSLIB_VERSION_MAJOR=8 -DWCSLIB_VERSION_MINOR=2 -I/Users/dev/Code/casacore -I/Users/dev/Code/casacore/build -I/opt/homebrew/include -I/Users/dev/Code/casacore/build/tables -Wextra -Wall -W -Wpointer-arith -Woverloaded-virtual -Wwrite-strings -pedantic -Wno-long-long -DNDEBUG -pthread -O3 -DNDEBUG -std=c++11 -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk -O3 -Wall -DNDEBUG -MD -MT tables/Dysco/CMakeFiles/tDysco.dir/tests/testdyscostman.cc.o -MF tables/Dysco/CMakeFiles/tDysco.dir/tests/testdyscostman.cc.o.d -o tables/Dysco/CMakeFiles/tDysco.dir/tests/testdyscostman.cc.o -c /Users/dev/Code/casacore/tables/Dysco/tests/testdyscostman.cc
/Users/dev/Code/casacore/tables/Dysco/tests/testdyscostman.cc:165:3: error: no type named 'directory_iterator' in namespace 'boost::filesystem'; did you mean 'std::__fs::filesystem::directory_iterator'?
  boost::filesystem::directory_iterator end_itr;
  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  std::__fs::filesystem::directory_iterator
/Library/Developer/CommandLineTools/SDKs/MacOSX14.4.sdk/usr/include/c++/v1/__filesystem/directory_iterator.h:39:7: note: 'std::__fs::filesystem::directory_iterator' declared here
class directory_iterator {
      ^
/Users/dev/Code/casacore/tables/Dysco/tests/testdyscostman.cc:167:27: error: no member named 'directory_iterator' in namespace 'boost::filesystem'
  for (boost::filesystem::directory_iterator itr("TestTable/"); itr != end_itr;
       ~~~~~~~~~~~~~~~~~~~^
```

The obvious solution is to downgrade boost, however this is not trivial on macos.

It seems like the issue is not present with boost 1.84

--- a/tables/Dysco/tests/testdyscostman.cc
+++ b/tables/Dysco/tests/testdyscostman.cc
@@ -1,5 +1,6 @@
 #include <boost/test/unit_test.hpp>
 #include <boost/filesystem/operations.hpp>
+#include <boost/filesystem/directory.hpp>
 
 #include <casacore/tables/Tables/ArrayColumn.h>
 #include <casacore/tables/Tables/Table.h>

